Geyser.Label rightClick menu

Edru2 opened the pull request:
#### Brief overview of PR changes/additions
Gives Geyser.Labels the possibility to have a right-click menu.
New functions:
```lua
testLabel:createRightClickMenu(cons)
```
Different cons option can style/size/format different levels of the menu.
The right click menu elements itself are given in a table of strings (**MenuItems**) which will be the names of the Labels.
Sub-tables of tables are then used to create the next menu level.
```lua 
testLabel:setMenuAction(name or parent.name as string, function, args)
 -- can be used the same way as setClickCallback
-- only one parent is needed and the name is then given like "parentname.elementname"
-- see example below
```

All the menu-labels are saved in ```testlabel.rightClickMenu.MenuLabels``` and subdirectories (for deeper levels) which can make them a bit difficult to access if special styling options per label or echoing alternative text and so on is needed.
Therefore it can be easier to use a new function to directly access the right-click menu labels by:
```lua
testLabel:findMenuElement(name or parent.name):myfunc(myargs)
```
See example below.

Additionally with this PR as Adjustable Container also use this right click menu it is now possible to change their right click menu style (darkmode or lightmode) by using:
```lua
myAdjustableContainer:changeMenuStyle(mode) -- mode is "dark" or "light"
```

#### Motivation for adding to Mudlet
Adjustable.Container has a right click menu for a long time now. I thought maybe other people want to use it as well.
And it seems also useful to standardize/simplify the way to make a right-click menu

#### Other info (issues closed, discussion etc)

(Ugly) example to show all the possibilities and different styling options.
```lua
testLabel = testLabel or Geyser.Label:new({name = "TestLabel", message = "Please right click!", format = "cb15", x=10, y=10, width ="30%", height = "15%", color = "purple"}, testAdjCont)
-- ugly showcase with different styles in different levels
-- the numbers show the deepnes of the level for example Style is the default style but then Style3 is for Labels in level 3 of deepness
testLabel:createRightClickMenu(
                              {MenuItems = {"This", "is",{"Level 2","Have",{"Fun"}}, "a", "rightClick",{"Menu"}}, 
                              Style = "Light", 
                              Style3 = "Dark", 
                              MenuWidth2 = 80, 
                              MenuWidth3 = 180, 
                              MenuFormat1 = "l10",
                              MenuStyle2 = [[QLabel::hover{ background-color: rgba(0,255,150,100%); color: white;} QLabel::!hover{color: brown; background-color: rgba(100,240,240,100%);} ]]}
                              )

-- use the label directly. This is the label with the name "right click"
testLabel.rightClickMenu.MenuLabels["rightClick"]:setAlignment("c")

-- use config label to get to the label settings. in this case the Label "Level2" 
-- its parent is the label "is"
testLabel:findMenuElement("is.Level 2"):echo([[<font size="5" face="Noto Emoji">⚓</font> Level 2]])

-- works like setClickCallback can use directly a function or give function and args by string
testLabel:setMenuAction("This", function() echo("Hello\n") end )

-- if a label has a parent it needs to be given in the name. divided by a .
testLabel:setMenuAction("rightClick.Menu", "display", "Test" )

-- no matter how deep the level is only one parent needs to be given
testLabel:setMenuAction("Have.Fun", function(event) echo("\nEVENT:") display(event) end )
```

Edru2 committed:
Geyser.Label rightClick menu

gives labels the possibility to have a right click menu

demonnic commented:
about to start testing this, but curious: is this what adjustable.containers uses? If so, should this PR also include changing Adjustable.Containers to make use of this instead? Keep it DRY and all that.

demonnic commented:
Quite like this, in my testing it seems to do what it says on the tin

Edru2 committed:
added rightclickmenu to adjustable.container

Edru2 commented:
I added the right click menu to adjustable container and added a few new functions like hide/show/add menu items and change item index (which where needed for adjustable containers). but I have to retest it therefore it's a draft for now

Edru2 committed:
darkmode for adjustable.container

Edru2 committed:
writing docs and got rid of configmenuLabel

Edru2 commented:
Ready for review (again) now. 

Please be aware that the example code changed a bit because after testing and using this a bit I noticed that a function was not really needed (configMenuLabel)

demonnic reviewed (approved):
![image](https://user-images.githubusercontent.com/3660/93634566-d2905c00-f9be-11ea-9baf-ecb0727e97ec.png)

test code works also.

demonnic commented:
Tested an adjustable container as well and find no regressions
